quic/s2n-quic-transport/src/stream/stream_container.rs

```diff
@@ -181,11 +185,7 @@ impl<S: StreamTrait> InterestLists<S> {
             ($interest:expr, $link_name:ident, $list_name:ident) => {
                 if $interest != node.$link_name.is_linked() {
                     if $interest {
-                        if matches!(result, StreamContainerIterationResult::Continue) {
-                            self.$list_name.push_back(node.clone());
-                        } else {
-                            self.$list_name.push_front(node.clone());
-                        }
+                        self.$list_name.push_back(node.clone());
```

Comment:
Was this logic only used for transmissions?

Reply:
No, this macro is used for all lists. Pushing to the front after an error doesn't work for the transmission list because if the first node fills up the packet then the second node will try to fill up the packet and then error (and push itself to the front.)
I guess I'm not sure if the logic I removed is important to the other lists. But if this branch is important I can add it back. I think I would need a separate macro for the transmission list in that case.

Reply:
That logic was added solely [for the transmission list.](https://github.com/aws/s2n-quic/pull/1069) Obviously it touched more than the transmission list, but I'm doubting it matters for the other lists. The fairness behavior for the transmission list is preserved, but the logic has moved into `send_on_transmission_list`